openquake/calculators/event_based_risk.py

```diff
@@ -639,3 +640,109 @@ def build_agg_curve_stats(self, builder, agg_curve, loss_curve_dt):
         self.datastore['agg_curve-stats'] = agg_curve_stats
         self.datastore['agg_curve-stats'].attrs['nbytes'] = (
             agg_curve_stats.nbytes)
+
+        
+def losses_by_taxonomy(riskinput, riskmodel, rlzs_assoc, assetcol, monitor):
+    """
+    :param riskinput:
+        a :class:`openquake.risklib.riskinput.RiskInput` object
+    :param riskmodel:
+        a :class:`openquake.risklib.riskinput.CompositeRiskModel` instance
+    :param rlzs_assoc:
+        a class:`openquake.commonlib.source.RlzsAssoc` instance
+    :param assetcol:
+        AssetCollection instance
+    :param monitor:
+        :class:`openquake.baselib.performance.Monitor` instance
+    :returns:
+        a numpy array of shape (T, L, R)
+    """
+    lti = riskmodel.lti  # loss type -> index
+    L, R = len(lti), len(rlzs_assoc.realizations)
+    T = len(assetcol.taxonomies)
+    taxonomy_id = {t: i for i, t in enumerate(sorted(assetcol.taxonomies))}
+    losses = numpy.zeros((T, L, R), F64)
+    for out in riskmodel.gen_outputs(riskinput, rlzs_assoc, monitor, assetcol):
+        t = taxonomy_id[out.assets[0].taxonomy]
```

Comment:
Please add a comment specifying that, by construction, all assets have the same taxonomy and the list is not empty